examples/RNOneSignal/App.js

```diff
@@ -8,484 +8,692 @@
+        OneSignal.init(appId, {
+            kOSSettingsKeyAutoPrompt: true,
+        });
+
+        // Notifications will display as NOTIFICATION type
+        OneSignal.inFocusDisplaying(2);
+
+        // If the app requires privacy consent check if it has been set yet
+        if (requiresPrivacyConsent) {
+        // async 'then' is only so I can sleep using the Promise helper method
+            OneSignal.userProvidedPrivacyConsent().then(async (granted) => {
+                // For UI testing purposes wait X seconds to see the loading state
+                await sleep(0);
+
+                console.log('Privacy Consent status: ' + granted);
+                this.setState({hasPrivacyConsent:granted, isPrivacyConsentLoading:false});
+            });
+        }
+
+        OneSignal.getPermissionSubscriptionState((response) => {
+            console.log('Device state:');
+            console.log(response);
+
+            let notificationsEnabled = response['notificationsEnabled'];
+            let isSubscribed = response['subscriptionEnabled'];
+
+            this.setState({isSubscribed:notificationsEnabled && isSubscribed, isSubscriptionLoading:false}, () => {
+                OneSignal.setSubscription(isSubscribed);
+            });
+        });
+
+        // Examples for using native IAM public methods
+//        this.oneSignalInAppMessagingExamples();
+
+        // Examples for using native Outcome Event public methods
+//        this.oneSignalOutcomeEventsExamples();
+
+    }
+
+    async componentDidMount() {
+        OneSignal.addEventListener('received', this.onNotificationReceived);
+        OneSignal.addEventListener('opened', this.onNotificationOpened);
+        OneSignal.addEventListener('ids', this.onIdsAvailable);
+//        OneSignal.addEventListener('subscription', this.onSubscriptionChange);
+//        OneSignal.addEventListener('permission', this.onPermissionChange);
+        OneSignal.addEventListener('emailSubscription', this.onEmailSubscriptionChange);
+        OneSignal.addEventListener('inAppMessageClicked', this.onInAppMessageClicked);
+    }
+
+    componentWillUnmount() {
+        OneSignal.removeEventListener('received', this.onNotificationReceived);
+        OneSignal.removeEventListener('opened', this.onNotificationOpened);
+        OneSignal.removeEventListener('ids', this.onIdsAvailable);
+//        OneSignal.removeEventListener('subscription', this.onSubscriptionChange);
+//        OneSignal.removeEventListener('permission', this.onPermissionChange);
+        OneSignal.removeEventListener('emailSubscription', this.onEmailSubscriptionChange);
+        OneSignal.removeEventListener('inAppMessageClicked', this.onInAppMessageClicked);
+    }
+
+    /**
+     Validate email method
+     */
+    validateEmail(email) {
+        var re = /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
```

Comment:
maybe name it regex?

Reply:
I didnt make this, and dont plan on touching it right now haha